api/client/service/update.go

```diff
@@ -249,7 +249,7 @@ func updatePlacement(flags *pflag.FlagSet, placement *swarm.Placement) {
 
 func updateLabels(flags *pflag.FlagSet, field *map[string]string) {
 	if flags.Changed(flagLabelAdd) {
-		if field == nil {
+		if *field == nil {
```

Comment:
Do we need this same fix below in `if field != nil` ?

If someone tries to remove a label from a service that has no labels yet.